tools/bapbundle.ml

```diff
@@ -189,7 +202,7 @@ module Remove = struct
   let args = [destdir_arg]
   let main () =
     if target.contents = "" then raise Target_unspecified;
-    let file = Filename.concat !destdir !target in
+    let file = Filename.concat !destdir !target |> rename_if_plugin in
```

Comment:
Try not to abuse the pipe operators (including `>>|` and `>>=`). These operators are used to build a pipeline of actions. For example:
```ocaml
let prepare file = 
    file |> move |> rename |> fix |> move ...
```

Notice, that all constituents of the pipeline, except the first one, are actions, i.e., verbs that are expected to have an effect. 

In our case, your usage of the pipeline looks justified, except that, as I explained above, the `rename_if_plugin` is not an action. So, the pipeline here just amplifies the delusion, that rename will actually rename the file. 

In fact, if we will use a proper naming discipline here, it would look like:

```ocaml
let file = path |> normalized 
```

Unless you're French this order is a little bit weird and the readability sucks, as `normalized path` is much better. But what is more significant, the absence of the pipeline operator tells us, that there are no side-effects here. 

If you just want to reduce the amount of parentheses, then I would suggest you either to use a normal application operator:

```ocaml
let file = Name.normalized @@ Filename.concat !destdir !target in
```

or multiple let bindings:

```ocaml
let file = Filename.concat !destdir !target in
let file = Name.normalized file in
...
```